seed/campaigns.ts

```diff
@@ -1,59 +1,143 @@
-import { Prisma, PrismaClient, User } from '@prisma/client'
+import {
+  CampaignUpdateHistoryType,
+  Prisma,
+  PrismaClient,
+  User,
+} from '@prisma/client'
 import { campaignFactory } from './factories/campaign.factory'
 import { campaignUpdateHistoryFactory } from './factories/campaignUpdateHistory.factory'
 import { userFactory } from './factories/user.factory'
 import { pathToVictoryFactory } from './factories/pathToVictory.factory'
 import { buildSlug } from 'src/shared/util/slug.util'
 import { getFullName } from 'src/users/util/users.util'
-
-const NUM_CAMPAIGNS = 20
+import { Campaign } from '@prisma/client'
+import fixedCampaigns from './fixedCampaigns.json'
+import 'prisma/schema/pathToVictory.jsonTypes'
+const NUM_GENERATED_CAMPAIGNS = 40
 const NUM_UPDATE_HISTORY = 3
+const FIXED_CAMPAIGNS: Partial<Campaign>[] = fixedCampaigns
+
+type CampaignUpdateHistory = {
+  id: number
+  createdAt: Date
+  updatedAt: Date
+  userId: number
+  campaignId: number
+  type: CampaignUpdateHistoryType
+  quantity: number
+}
+
+type FakeP2V = {
+  id: number
+  createdAt: Date
+  updatedAt: Date
+  campaignId: number
+  data: PrismaJson.PathToVictoryData
+}
 
 export default async function seedCampaigns(
   prisma: PrismaClient,
   existingUsers: User[],
 ) {
-  const fakeP2Vs: any[] = []
-  const fakeUpdateHistory: any[] = []
-
+  const fakeP2Vs: FakeP2V[] = []
+  const fakeUpdateHistory: CampaignUpdateHistory[] = []
   const campaignIds: number[] = []
 
-  for (let i = 0; i < NUM_CAMPAIGNS; i++) {
-    let user = existingUsers[i]
-    if (!user) {
-      const userData = userFactory()
-      user = await prisma.user.create({
-        data: {
-          ...userData,
-          metaData:
-            userData.metaData !== null ? userData.metaData : Prisma.JsonNull,
-        },
-      })
-    }
-    const campaign = await prisma.campaign.create({
-      data: campaignFactory({
-        userId: user.id,
-        slug: buildSlug(getFullName(user)),
-      }),
-    })
-
-    campaignIds.push(campaign.id)
-    fakeP2Vs.push(pathToVictoryFactory({ campaignId: campaign.id }))
+  const loopLength = Math.max(FIXED_CAMPAIGNS.length, NUM_GENERATED_CAMPAIGNS)
 
-    for (let j = 0; j < NUM_UPDATE_HISTORY; j++) {
-      fakeUpdateHistory.push(
-        campaignUpdateHistoryFactory({
-          campaignId: campaign.id,
-          userId: user.id,
-        }),
+  for (let i = 0; i < loopLength; i++) {
+    if (i < FIXED_CAMPAIGNS.length) {
+      const creationData = await createCampaignAndUser(
+        existingUsers,
+        prisma,
+        FIXED_CAMPAIGNS[i],
       )
+      const { campaignId, fakeP2V, fakeUpdateHistory } = creationData
```

Comment:
You can do this destructuring on the same line as the function call above. No need for the extra variable.